patroni/postgresql.py

```diff
@@ -189,25 +189,78 @@ def sync_from_leader(self, leader):
+        connstring = leader.conn_url
+        # get list of replica methods from config
+        replica_list = self.config.get('create_replica_method', 'basebackup')
+        replica_methods = [rm.strip() for rm in replica_list.split(',')]
+        # go through them in priority order
+        for replica_method in replica_methods:
+            # if the method is basebackup, then use the built-in
+            if replica_method == "basebackup":
+                ret = self.basebackup(leader, env)
+                if ret == 0:
+                    # if basebackup succeeds, exit with success
+                    return 0
+            else:
+                # user-defined method; check for configuration
+                # not required, actually
+                if replica_method in self.config:
+                    # look to see if the user has supplied a full command path
+                    # if not, use the method name as the command
+                    if "command" in self.config[replica_method]:
+                        cmd = self.config[replica_method]["command"]
+                    else:
+                        cmd = replica_method
+
+                    # get the rest of the replica config
+                    method_config = self.config[replica_method].copy()
+                    # remove the command and turn it into a shlex set
+                    del method_config["command"]
+                    # add the default parameters
+                    method_config.update({"scope": self.scope,
+                                          "role": "replica",
+                                          "datadir": self.data_dir,
+                                          "connstring": connstring})
+                    params = ["--{0}={1}".format(arg, val) for arg, val in method_config.items()]
+                else:
+                    cmd = replica_method
+                    method_config = {"scope": self.scope,
+                                    "role": "replica",
+                                    "datadir": self.data_dir,
+                                    "connstring": connstring}
 
-    def create_replica(self, master_connection, env):
-        self.set_state('building replica from {host}:{port}'.format(**master_connection))
-        connstring = self.build_connstring(master_connection)
-        cmd = self.config['restore']
-        try:
-            ret = subprocess.call(shlex.split(cmd) + [self.scope, "replica", self.data_dir, connstring], env=env)
-            self.delete_trigger_file()
-        except:
-            logger.exception('Error when creating replica')
-            ret = 1
-        if ret != 0:
-            self.set_state('failed to build replica from {host}:{port}'.format(**master_connection))
-        return ret
+                try:
+                    # call script with the full set of parameters
+                    ret = subprocess.call(shlex.split(cmd) + shlex.split(method_config), env=env)
```

Comment:
The shlex.split(method_conifg) will raise an exception, since method_config is a dictionary, and shlex.split expects a string. Since you are declaring the 'params' variable, but not using them anywhere, my guess is that you originally prepared them to be passed into subprocess.call.